nemo/collections/asr/modules/conformer_encoder.py

```diff
@@ -671,14 +671,15 @@ def forward_internal(
             return audio_signal, length
 
     def update_max_seq_length(self, seq_length: int, device):
+        # TODO: this sync seems unnecessary, remove or disable properly later
```

Comment:
@zhehuaichen what should we do about this check? It was commented out for practically all of our speechllm experiments and we never had an issue because of that. OTOH when it's not commented out, we risk running into TP deadlocks. Should we just remove this or add an option?

Reply:
This is necessary if your audio samples increase in length than the max length during training -which is all non speech lm training. Keep it. If you want to avoid hitting this, setup a max sequence length for conformer that is larger than the largest it will ever see during training - then this path is skipped entirely by default.

Reply:
> your audio samples increase in length than the max length during training -which is all non speech lm training.

The samples increase in size during speech lm training as well. That's why I'm saying this check is not necessary at all. I am aware it fixed some issue in the past, but it looks like this solution must have been masking some other issue that is no longer present in the codebase.

> then this path is skipped entirely by default.

The issue is the `if` condition: it performs an NCCL sync which prevents us from using model parallelism.

Reply:
Talked with @titu1994 : we'll add a flag here

Reply:
thanks